projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -40,34 +44,55 @@ const ADD_MEDIA_LABEL = __( 'Set Social Image', 'jetpack' );
 const REPLACE_MEDIA_LABEL = __( 'Replace Social Image', 'jetpack' );
 const REMOVE_MEDIA_LABEL = __( 'Remove Social Image', 'jetpack' );
 
-const ALLOWED_MEDIA_TYPES = [ 'image/jpeg', 'image/png' ];
-
 /**
  * Wrapper that handles media-related functionality.
  *
  * @returns {object} The media section.
  */
 export default function MediaSection() {
+	const [ validationError, setValidationError ] = useState( null );
 	const { attachedMedia, updateAttachedMedia } = useAttachedMedia();
+	const { connections } = useSocialMediaConnections();
+	const enabledConnections = connections.filter( connection => connection.enabled );
+
+	const { maxImageSize, allowedImageTypes, getValidationError } = useMediaRestrictions(
+		enabledConnections
+	);
 
 	const mediaObject = useSelect( select =>
 		select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } )
 	);
-
 	const { mediaWidth, mediaHeight, mediaSourceUrl } = getMediaDetails( mediaObject );
 
+	useEffect( () => {
+		// Removes selected media if connection change results in invalid image
+		if ( ! mediaObject ) {
+			return;
+		}
+		const error = getValidationError( mediaObject.media_details?.filesize, mediaObject.mime_type );
+		if ( error ) {
+			setValidationError( error );
+			updateAttachedMedia( [] );
+		}
+	}, [ updateAttachedMedia, mediaObject, getValidationError ] );
```

Comment:
If we get the filesize and mimetype coming back from `getMediaDetails`, then we could list them here, so it's a primitive type - even though I think `mediaObject`'s reference shouldn't be changing.